snapshot/src/test/java/io/zeebe/snapshots/impl/FileBasedSnapshotStoreTest.java

```diff
@@ -134,51 +125,45 @@ public void shouldLoadLatestSnapshotWhenMoreThanOneExistsAndDeleteOlder() throws
         });
 
     // when
-    try (final ActorScheduler otherScheduler = ActorScheduler.newActorScheduler().build()) {
-      otherScheduler.start();
-
-      final var snapshotStore =
-          new FileBasedSnapshotStoreFactory(otherScheduler, 1)
-              .createReceivableSnapshotStore(root.toPath(), partitionId);
-
-      // then
-      final var currentSnapshotIndex = snapshotStore.getCurrentSnapshotIndex();
-      assertThat(currentSnapshotIndex).isEqualTo(10L);
-      // should delete older snapshots
-      assertThat(snapshotsDir.toFile().list())
-          .hasSize(1)
-          .containsExactly(snapshotStore.getLatestSnapshot().orElseThrow().getId());
-    }
+    constructableSnapshotStore.close();
+    final var snapshotStore =
+        new FileBasedSnapshotStoreFactory(scheduler.get(), 1)
+            .createReceivableSnapshotStore(root.toPath(), partitionId);
+
+    // then
+    assertThat(snapshotStore.getCurrentSnapshotIndex()).isEqualTo(10L);
+    assertThat(snapshotsDir.toFile().list())
```

Comment:
```suggestion
    assertThat(snapshotsDir).isDirectoryContaining(p -> p.equals(snapshotStore.getLatestSnapshot().map(PersistedSnapshot::getPath).orElseThrow());
```

I didn't test this so maybe don't commit directly, but you can use more specific assertions to test paths. I recommend looking into them, and using the IntelliJ plugin Cajon which also helps find some out :slightly_smiling_face:

Reply:
It can be made more readable, e.g.

```java
final var expectedSnapshotPath = snapshotStore.getLatestSnapshot().map(PersistedSnapshot::getPath).orElseThrow();
assertThat(snapshotsDir).isDirectoryContaining(p -> p.equals(expectedSnapshotPath);
```

The nice thing here is the error message - one tells you that you have the wrong paths on the file system (and what's expected), and the other just compares two strings in a vacuum. 

Admittedly adding an assertion message mitigates the issue, but I find having the more precise assertions (as possible) typically helps figure out what went wrong faster. Here we compare paths directly, in the other we compare strings (with the actual one having originated from a path).

I wouldn't insist too much though, but I just want us to get in the mindset that for test/assertions, it's not just how readable it is when reading the code, but also about the error produced by it when it fails.

Reply:
Actually we have to use `isDirectoryNotContaining`. `isDirectoryContaining` passes if there is atleast one file that matches the predicate.

Reply:
Sorry, I understood it would pass if all matched, so it could be used as an exactly.

Reply:
I think it is fine with the inverse comparison. You get the following error if it fails
```
Expecting directory:
  /tmp/junit4691337748246145506/snapshots/pending
not to contain any files matching the given filter but found some:
 [/tmp/junit4691337748246145506/snapshots/pending/10-1-10-10]
```